.github/workflows/build.yaml

```diff
@@ -0,0 +1,63 @@
+name: "Build For Algojig"
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+  workflow_dispatch:
```

Comment:
this can be tweaked if needed, leaving this in-place will allow you to trigger it manually, figured it will be useful for quick tests as this pipeline is still rather an initial draft setup